### ISSUE TITLE
NPE when bridge doesn't return sensors

### ISSUE THREAD

**Issue by vmsh0** [ZeroOne3010/yetanotherhueapi#40]
Hi,

You have an NPE when the bridge doesn't send the "sensors" object entirely and the library refreshes the cached data:

```
Exception in thread "Thread-3" java.lang.NullPointerException
	at io.github.zeroone3010.yahueapi.Hue.refresh(Hue.java:151)
	at io.github.zeroone3010.yahueapi.Hue.doInitialDataLoadIfRequired(Hue.java:116)
	at io.github.zeroone3010.yahueapi.Hue.getGroupsOfType(Hue.java:179)
	at it.uniupo.giiot.iot.plugins.feedback.SmartLightsFeedback.onChange(SmartLightsFeedback.java:42)
	at it.uniupo.giiot.iot.core.AulosIotSubsystem.run(AulosIotSubsystem.java:205)
```

The Hue emulator we're using does that and it can't be worked around as sensors are not supported.

Since we have to deliver in just a few days, I've deployed a dirty fix on our side. If you're interested, I could clean it up and send a PR. (basic idea: check whether `Root#getSensors()` is null; if it is, fall back to an empty `Map` instead of trying to process it.)

**Comment by vmsh0** [comment]
Should you be interested:

https://github.com/ZeroOne3010/yetanotherhueapi/compare/master...vmsh0:master

**Comment by ZeroOne3010** [comment]
Hi, please do send that PR! :)

**Comment by ZeroOne3010** [comment]
OK never mind, I implemented the NPE protection to that part and the other similar parts myself. 😄

**Comment by ZeroOne3010** [comment]
The fix has now been published in version 2.2.0, please find it from the public Maven repository as usual:

```xml
<dependency>
    <groupId>io.github.zeroone3010</groupId>
    <artifactId>yetanotherhueapi</artifactId>
    <version>2.2.0</version>
</dependency>
```

Thank you for reporting the issue! 👍

**Comment by vmsh0** [comment]
Thank you!Il 15 mar 2021 11:34, Ville Saalo ***@***.***> ha scritto:

The fix has now been published in version 2.2.0, please find it from the public Maven repository as usual:

<dependency>

    <groupId>io.github.zeroone3010</groupId>

    <artifactId>yetanotherhueapi</artifactId>

    <version>2.2.0</version>

</dependency>

Thank you for reporting the issue! 👍
